### PR TITLE
rabbitmq: Allow to use rabbitmqctl

### PR DESCRIPTION
**Pull request by sileht**

### CODE REVIEW

File: pifpaf/drivers/rabbitmq.py [hunk 1]
```diff
@@ -50,57 +52,63 @@ def get_parser(cls, parser):
...
             "RABBITMQ_PIDFILE": os.path.join(self.tempdir, nodename, "pid"),
         }
-        complete_env.update(env)
+        complete_env.update(self.env)
         c, _ = self._exec(["rabbitmq-server"], env=complete_env,
                           path=self._path,
                           wait_for_line=b"Starting broker... completed")
-        self.addCleanup(self._kill, c.pid)
+        self.addCleanup(self.kill_node, nodename)
+        self._process[nodename] = c
```
Comment (jd) [inline pifpaf/drivers/rabbitmq.py]: Maybe add a check that there's not already nodename in the _process, ya never know.

File: pifpaf/drivers/rabbitmq.py [hunk 2]
```diff
@@ -50,57 +52,63 @@ def get_parser(cls, parser):
...
-        self._exec(["rabbitmqctl", "-n", nodename, "stop_app"],
-                   path=self._path, env=env)
-        self._exec(["rabbitmqctl", "-n", nodename, "join_cluster",
-                    master],
-                   path=self._path, env=env)
-        self._exec(["rabbitmqctl", "-n", nodename, "start_app"],
-                   path=self._path, env=env)
+    def kill_node(self, nodename, signal=signal.SIGTERM):
+        try:
+            self._kill(self._process[nodename].pid, signal=signal)
```
Comment (jd) [inline pifpaf/drivers/rabbitmq.py]: That can raise a KeyError too, you better catch that.

File: pifpaf/drivers/rabbitmq.py [hunk 3]
```diff
@@ -50,57 +52,63 @@ def get_parser(cls, parser):
...
+            os.waitpid(self._process[nodename].pid, 0)
+        except OSError:
+            pass
+
+    def join_cluster(self, nodename, master):
+        self.rabbitmqctl(nodename, ["stop_app"])
```
Comment (jd) [inline pifpaf/drivers/rabbitmq.py]: Check nodename validity?

**Comment by jd** [comment]
You should reorder your patches too, so the tests pass.

File: pifpaf/drivers/rabbitmq.py (hunk 3 above)
Comment (sileht) [inline pifpaf/drivers/rabbitmq.py]: Done

File: pifpaf/drivers/rabbitmq.py (hunk 2 above)
Comment (sileht) [inline pifpaf/drivers/rabbitmq.py]: Done

File: pifpaf/drivers/rabbitmq.py (hunk 1 above)
Comment (sileht) [inline pifpaf/drivers/rabbitmq.py]: Done

**Pull request merged**